Automattic/newspack-theme
# WooCommerce: make sure WC pages are picking up custom fonts/colour

### All Submissions:


### Changes proposed in this Pull Request:

This PR removes the Sass variables being set in the WooCommerce.scss file, and moves the typography selectors to the main theme SCSS files instead, to help the WooCommerce pages pick up the correct colours/fonts. As part of that, it removes a function that should have been applying the theme's custom colours, but it wasn't really working.

It also corrects a `.button` hover colour issue that was affecting all of the buttons on the WooCommerce pages (once a link was visited, it hovered dark text on a dark background, instead of white text on a dark background). 

**Future improvement:** Right now, the CSS generated for each child theme uses a combination of the parent theme's Sass, with some variables overwritten in each child theme (ones you can technically override include fonts face and style, colours, spacing, though we just do font face). The WooCommerce file is built a bit differently, and only uses the Sass variables from the parent theme. This is not a big deal right now -- only the colour variables are used in the WooCommerce Sass file, and they aren't changed from child theme to child theme yet, but it would be good to get this working if that changed down the road. The WooCommerce pages should still pick up the custom colours set in the Customizer.  

### How to test the changes in this Pull Request:

1. On a test site using custom fonts and colours, click through the WooCommerce purchase flow (shop -> product -> adding to cart -> checking out); while doing that, note the spots where the pages use the default fonts or colours. Some examples include:
* **Shop:** breadcrumb, title, prices use the wrong font; 'Sale' tags use the wrong font and colour
* **Product:** breadcrumb, price, tabs, description use the wrong fonts; 'Sale' tags use the wrong font and colour
* **Checkout:** coupon message uses the wrong fonts
2. Apply the PR and run `npm run build`
3. Confirm that the above pages are now using your custom fonts and colours

**Shop - before:**

![image](https://user-images.githubusercontent.com/177561/108290937-412c3e00-7146-11eb-946c-650bb7d5aa71.png)

**Shop - after:**

![image](https://user-images.githubusercontent.com/177561/108291121-923c3200-7146-11eb-9235-63567d5231a7.png)

**Product - before**

![image](https://user-images.githubusercontent.com/177561/108290999-5d2fdf80-7146-11eb-89b7-032b4c9dba92.png)

**Product - after**

![image](https://user-images.githubusercontent.com/177561/108291199-b39d1e00-7146-11eb-967c-291bcb9c2a76.png)

**Checkout - before**

![image](https://user-images.githubusercontent.com/177561/108291030-69b43800-7146-11eb-80bb-b3f006054589.png)

**Checkout - after**

![image](https://user-images.githubusercontent.com/177561/108291079-82245280-7146-11eb-9843-e4eb21c05540.png)

4. Try removing your custom fonts and colours, and pick a child theme (anything but Newspack default); confirm that the child theme's correct fonts (and not the default Newspack's theme's fonts). Good ones to check are Newspack Nelson, Katharine or Joseph, because they don't use the same body font (Georgia). 

### Other information:
